lib/trip_dispatcher.rb

```diff
@@ -67,6 +75,63 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    # Load the Drivers from the support/drivers.csv file and
+    # return a collection of Driver instances, note that drivers
+    # can be passengers too! Replace the instance of User in the
+    # passengers array with a corresponding instance of Driver
+    def load_drivers(filename)
+      # initialize drivers
+      drivers = []
+      # parse csv for drivers
+       CSV.open(filename, 'r', headers: true).map do |line|
+        id = line[0].to_i
+        passenger = find_passenger(line[0].to_i)
+        line[1].length
+
+        driver_data = Driver.new({id: line[0].to_i,
+          name: passenger.name,
+          phone_number: passenger.phone_number,
+          trips: passenger.trips,
+          vehicle_id: line[1],
+          status: line[2]})
+          drivers << driver_data
+      end
+      return drivers
+    end
+
+    def find_driver(id)
+      check_id(id)
+      return @drivers.find { |driver| driver.id == id }
+    end
+
+    def request_trip(user_id)
```

Comment:
This method is well done, but it's a bit long.  You could break it into some helper methods doing the tasks of:

1.  Selecting an available driver
1.  Creating a new trip